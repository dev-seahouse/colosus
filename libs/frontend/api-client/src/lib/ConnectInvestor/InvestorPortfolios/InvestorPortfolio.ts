import ConnectInvestorBaseApi from '../_Base/Base';
import type { ConnectPortfolioDetailsDto } from '@bambu/shared';

export type GetInvestorPortfolioByIdType = {
  id: string;
};

export type ConnectInvestorGetInvestorPortfolioDetailsResponseDto =
  ConnectPortfolioDetailsDto.IConnectPortfolioDetailsDto;

export class ConnectInvestorPortfolioDetailsApi extends ConnectInvestorBaseApi {
  // TODO: This BE API is not yet set up
  constructor(private readonly apiPath = '/investor-portfolios') {
    super();
  }

  /**
   * Gets the portfolio details for a specified portfolio for an investor.
   */
  public async getInvestorPortfolio(req: GetInvestorPortfolioByIdType) {
    return await this.axios.get<ConnectInvestorGetInvestorPortfolioDetailsResponseDto>(
      // TODO: This is not yet set up on the BE
      `${this.apiPath}/${req.id}/details`
    );
  }
}

export default ConnectInvestorPortfolioDetailsApi;
