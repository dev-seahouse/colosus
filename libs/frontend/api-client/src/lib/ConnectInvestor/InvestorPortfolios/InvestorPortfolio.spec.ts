import { describe } from 'vitest';
import { ConnectInvestorPortfolioDetailsApi } from './InvestorPortfolio';

describe('InvestorPortfolio', () => {
  const connectInvestorPortfolioApi = new ConnectInvestorPortfolioDetailsApi();

  describe('getModelPortfoliosSummary', () => {
    it('should return correct response', async () => {
      const res = await connectInvestorPortfolioApi.getInvestorPortfolio({
        id: 'id',
      });
      expect(res.data).toMatchSnapshot();
    });
  });
});
