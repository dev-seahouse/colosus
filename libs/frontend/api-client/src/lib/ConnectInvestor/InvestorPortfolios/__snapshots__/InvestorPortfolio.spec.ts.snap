// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InvestorPortfolio > getModelPortfoliosSummary > should return correct response 1`] = `
{
  "description": "This portfolio primarily looks to preserve capital through a conservative asset allocation.",
  "expectedReturnPercent": "4",
  "expectedVolatilityPercent": "3.5",
  "factSheetLink": "https://google.com",
  "id": "ff2693ce-7773-4bdf-913d-5f042b9c5893",
  "key": "CONSERVATIVE",
  "name": "Conservative Portfolio",
  "products": [
    {
      "assetClass": "MF",
      "assetCurrentValue": 7200,
      "assetUnits": 4.5,
      "name": "AGF globalsustainable Growth Fund",
      "ticker": "IWDA",
    },
    {
      "assetClass": "MF",
      "assetCurrentValue": 7200,
      "assetUnits": 4.5,
      "name": "Quantified Alternative Investment",
      "ticker": "VOO",
    },
    {
      "assetClass": "MF",
      "assetCurrentValue": 7200,
      "assetUnits": 4.5,
      "name": "Quantified Managed Income",
      "ticker": "IWDA",
    },
    {
      "assetClass": "MF",
      "assetCurrentValue": 103,
      "assetUnits": 0.21,
      "name": "Tradeweb for US Government Securitiesreate Bond ETF - USD",
      "ticker": "SPX",
    },
    {
      "assetClass": "Cash",
      "assetCurrentValue": 100,
      "name": "US Dollar Holdings in Portfolio",
    },
  ],
  "riskProfileId": "b6739bd8-6267-4bf7-bb1f-e112c54bec71",
}
`;
